### PR TITLE
feat(test-runner): multiple WebServer configs

### LINKED ISSUE #8206: [Feature]: Support launching multiple web servers
**Context:**
- GOOD Playwright Version: 1.13
- BAD Playwright Version: 1.14
- Operating System: Mac (didn't test other OS)
- Extra: N/A

**Code Snippet**

This used to work on 1.13 (updated to 1.14 option names). 

```javascript
const config: PlaywrightTestConfig = {
  webServer: [
    {
      command: 'yarn workspace @app/graphql start',
      port: 4000, 
      timeout: 60000,
      reuseExistingServer: !isCI,
    },
    {
      command: 'yarn workspace @app/frontend start',
      port: 3000, 
      timeout: 60000,
      reuseExistingServer: !isCI,
    }
  ]
};
```

**Describe the bug**

Previously we had the experimental `_launch` option to start a server for tests, and this options accepted an array of several servers to start. With 1.14 released it was renamed to `webServer` and only takes one server as a parameter. Maybe it's a weird use case but it was really helpful to be able to lift several servers. `baseUrl` fixture took the value of the last server launched.

Would it be possible to add this back?

### PR DESCRIPTION
**Pull request by rwoll**
Resolves #8206.

### CODE REVIEW

File: packages/playwright-test/src/webServer.ts [hunk 1]
```diff
@@ -38,6 +38,24 @@ export class WebServer {
...
 
+  public static async createAll(configs: WebServerConfig[], reporter: Reporter): Promise<() => Promise<void>> {
+    const webServers: WebServer[] = [];
+    const killAll =  async () => {
+      for (const server of webServers)
+        await server.kill();
+    };
+
+    try {
+      for (const config of configs)
```
Comment (rwoll) [inline packages/playwright-test/src/webServer.ts:49]: NB: In the future, we could get fancy and start them all at once, but reporting and teardown are a bit ugglier for this optimization.

File: packages/playwright-test/src/runner.ts [hunk 2]
```diff
@@ -448,7 +448,7 @@ export class Runner {
     await this._runAndReportError(async () => {
       for (const internalGlobalSetup of this._internalGlobalSetups)
         internalGlobalTeardowns.push(await internalGlobalSetup());
-      webServer = config.webServer ? await WebServer.create(config.webServer, this._reporter) : undefined;
```
Comment (mxschmitt) [inline packages/playwright-test/src/runner.ts:451]: why not create a WebServer multiplexer aka. WebServers?

Comment (rwoll) [reply packages/playwright-test/src/runner.ts:451]: Felt pre-mature given the impl without the class is just a few lines: https://github.com/microsoft/playwright/pull/13007/files#diff-acef657c29c494a2b26e391d22e5efec3513e9273e465fceb61e62d5cd91b264R41

File: packages/playwright-test/types/test.d.ts [hunk 3]
```diff
@@ -730,8 +731,33 @@ interface TestConfig {
    * });
    * ```
    *
+   * You can also use multiple servers:
+   *
+   * ```ts
+   * // playwright.config.ts
+   * import { PlaywrightTestConfig } from '@playwright/test';
+   * const config: PlaywrightTestConfig = {
+   *   webServer: [
```
Comment (aslushnikov) [inline packages/playwright-test/types/test.d.ts:740]: i wish it'd be called 'services'! Like [Github Actions services](https://docs.github.com/en/actions/using-containerized-services/about-service-containers)

**Review by aslushnikov — APPROVED** [review]
lgtm

Don't forget to add to "important developments"!

**Comment by rwoll** [comment]
Closing; we are going to experiment with plugins.

**Pull request closed without merging**